ios/imagemounter/personalized_image_mounter.go

```diff
@@ -0,0 +1,196 @@
+package imagemounter
+
+import (
+	"fmt"
+	"github.com/Masterminds/semver"
+	"github.com/danielpaulus/go-ios/ios"
+	log "github.com/sirupsen/logrus"
+	"io"
+	"os"
+	"path"
+)
+
+type PersonalizedDeveloperDiskImageMounter struct {
+	deviceConn ios.DeviceConnectionInterface
+	plistRw    ios.PlistCodecReadWriter
+	version    *semver.Version
+	tss        tssClient
+	ecid       uint64
+}
+
+func NewPersonalizedDeveloperDiskImageMounter(entry ios.DeviceEntry, version *semver.Version) (PersonalizedDeveloperDiskImageMounter, error) {
+	values, err := ios.GetValuesPlist(entry)
+	if err != nil {
+		return PersonalizedDeveloperDiskImageMounter{}, nil
+	}
+	var ecid uint64
+	if e, ok := values["UniqueChipID"].(uint64); ok {
+		ecid = e
+	} else {
+		return PersonalizedDeveloperDiskImageMounter{}, fmt.Errorf("could not get ECID from device")
+	}
+	deviceConn, err := ios.ConnectToService(entry, serviceName)
+	if err != nil {
+		return PersonalizedDeveloperDiskImageMounter{}, err
```

Comment:
same here, please start wrapping errors like above. I recently learned that this is how you do it so we can later properly debug things when they fail. 
fmt.Errorf("NewPersonalizedDeveloperDiskImageMounter: failed connecting to serviceName with err: %w", err)